### ISSUE TITLE
help wanted - Identifier converter to update MSB Para records

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#8343]
**Describe the bug**
I tried to use the Identifier converted for the first time to fix some records that were entered as institutional catalog numbers instead of Arctos record guids.  https://arctos.database.museum/loaders/identifierConverter.cfm?action=ld
First I downloaded the template for this loader and used it to create the attached file.
I uploaded the file to the loader, marked to autoload, and received the "lookup success" status; I then changed the status to autoload_passthrough so it would go through the identifier loader and automatically update the records.

The results were not as I expected. I was attempting to convert an existing identifier from type= institutional catalog number to type = Arctos record guid with the same issued by. I used the full arctos url for the new identifier value, and Arctos record guid for the new identifier type. 
Instead, the record shows that a entirely new identifier was added instead of correcting the old identifier as expected. Furthermore, the new identifier uses the old identifier type = institutional catalog number instead of the specified new value of Arctos record guid. It uses the old identifier value instead of the new full url of the Arctos record guid. And it does not provide the issued by agent. All of these values were in my input file, but none were successfully uploaded or even updated. 

 For example: https://arctos.database.museum/guid/MSB:Para:45863

**Screenshots**
.
![Screenshot 2024-11-27 11 05 55](https://github.com/user-attachments/assets/03db2ac9-bb73-49ea-b0e4-ef05895526e5)

**Data**

[identifierDownload_MSBPara_Herp_institutionalcatnum.csv](https://github.com/user-attachments/files/17938505/identifierDownload_MSBPara_Herp_institutionalcatnum.csv)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome, safari] chrome

**Additional context**
If I have entered values in the file incorrectly, please provide the correct format and an explanation of how we can make this process more transparent and clear to users.

**Comment by dustymc** [comment]
I'm not quite sure how to respond to this. That tool won't do what I think you're trying to do, neither will the second tool, and I can't quite see where anything has suggested that either of them would. We probably need better documentation, but I'm not sure how to write it.

I'm also wondering if this tool isn't just an unnecessary complexity that should be removed at this point, I think these 'legacy' identifiers only exist because they've been recently created, and the other use cases of which I'm aware are pretty easy to address in a spreadsheet-or-whatever.

> tried to use the Identifier converted for the first time to fix some records that were entered as institutional catalog numbers 

It will just return whatever you input for that type, the tool cannot do anything useful from that starting point.

> First I downloaded the template for this loader and used it to create the attached file.

I suppose the input/output terminology in the instructions isn't clear, but I'm not sure how to do better. In any case, the attached file is not the kind of input that's expected by this tool.

>received the "lookup success"

"found something" (what I suspect happened, but you haven't shared those data so I have to guess) and "did whatever you were expecting" (?) are not the same thing. It should have just overwritten your inputted-output fields with your inputs in this situation - "Use this only in simple situations, and only after testing with the first option." 

>so it would go through the identifier loader and automatically update the records.

That tool could definitely use some help-help as well, but I'd think the title would suggest the functionality sufficiently: Bulkload Identifiers/Relationships.

**Comment by campmlc** [comment]
So  if this tool doesn't do what it says it does in the title 'identifier converter ' and instead just loads new identifiers, it is definitely of no use and should be removed. 
So to fix what I just did, can you please unload the duplicate institutional catalog number identifiers I loaded yesterday using the Arctos-dupplied incorrect template?

**Comment by campmlc** [comment]
FWIW there are 14 records in the identifier converter with status lookup success entered by @dustymc. The data appear to be on exactly the same format as the file I uploaded using the template I downloaded from this tool. Perhaps a good test would be to mark these as autoload_, passthrough following the tool instructions ( status option 2). This is should replicate exactly what I did with the attached file, except I was converting to type =Srctos record guid instead of identifier and GenBank.

**Comment by mkoo** [comment]
HI Mariel, I see two actions based on your issue: a) better wording on the tool's page, which I can work on with others. b) you can have Dusty just add these 32 records with the desired identifier for you as well as then remove the dup identifiers.

 It's still a two part action, which needs to be better documented: add desired identifier (bulkload tool) then removing the old ones (unload tool). 

So let Dusty know about action b (are these the only identifers to add-- in your CSV attached in the original issue?) Thanks

**Comment by campmlc** [comment]
Yes, better documentation is needed as the current documentation claims to do something it does not, at least as far as is decipherable. 
And yes, I request that @dustymc add the desired identifier in the csv and remove the incorrect institutional catalog numbers that lack the Arctos guid type, format, and correct issued by.

**Comment by dustymc** [comment]
>add the desired identifier in the csv and remove the incorrect

done
